rust/agama-software/src/model.rs

```diff
@@ -86,7 +64,7 @@ pub trait ModelAdapter: Send + Sync + 'static {
     ) -> Result<(), service::Error>;
 
     /// Probes system and updates info about it.
-    async fn probe(&mut self, product: &ProductSpec) -> Result<(), service::Error>;
+    async fn probe(&mut self) -> Result<(), service::Error>;
```

Comment:
why you remove parameter here? Will model hold the product specification?

Reply:
The probe action has not been implemented yet, but the idea is that you the product is set only in `SetConfig`. The probe action is only to refresh the respositories. And if we need the product spec for some reason, then yes, I guess we will need to keep it.

Reply:
well, it will mean that for setting product you need
1. set product
2. add product repos
3. refresh repos
4. add product itself and all its mandatory patterns and packages.

if this workflow is ok for you, then it can be done without product spec....but in that case it will need at least update comment, maybe even better to update method name to something like refresh_repositories

Reply:
Then let's rename it.